rastervision_core/rastervision/core/data/dataset_config.py

```diff
@@ -26,6 +32,9 @@ def update(self, pipeline=None):
             for s in self.test_scenes:
                 s.update(pipeline=pipeline)
 
+        self.img_channels = len(
+            self.train_scenes[0].raster_source.channel_order)
```

Comment:
If we throw an error on `channel_order` being `None`, it will effectively make it non-optional, which I don't think is desirable. And in the absence of `channel_order`, there is also no way to determine it from the config alone, without building a scene first. 

I have therefore modified `SemanticSegmentation(RVPipeline)` so that if `DatasetConfig.img_channels` is not set, it builds the first scene to extract `.raster_source.num_channels` and sets `DatasetConfig.img_channels` to it. It then also calls `.update()` and `.validate()` on it for good measure.

I think this is a good solution that does not disturb anything else.